packages/labs/ssr/src/lib/render-value.ts

```diff
@@ -829,4 +920,66 @@ function* renderAttributePart(
   }
 }
 
+/**
+ * Returns a debug string suitable for an error message describing a
+ * TemplateResult.
+ */
+function displayTemplateResult(
+  result: TemplateResult | CompiledTemplateResult
+) {
+  if (isCompiledTemplateResult(result)) {
+    return result._$litType$.h.join('${...}');
+  }
+  return result.strings.join('${...}');
+}
+
 const getLast = <T>(a: Array<T>) => a[a.length - 1];
+
+function willExecute(node: Element | Template): boolean {
```

Comment:
Should some assumptions be documented on this function. E.g., this only holds true on server templates?

Reply:
Yes, and thinking about it, a more specific name is called for as well.